pkg/apis/pingcap/v1alpha1/tidbautoscaler_types.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"k8s.io/api/autoscaling/v2beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +k8s:openapi-gen=true
+// TidbClusterAutoScaler is the control script's spec
+type TidbClusterAutoScaler struct {
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec describes the state of the TidbClusterAutoScaler
+	Spec TidbClusterAutoScalerSpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +k8s:openapi-gen=true
+// TidbClusterAutoScalerList is TidbClusterAutoScaler list
+type TidbClusterAutoScalerList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ListMeta `json:"metadata"`
+
+	Items []TidbClusterAutoScaler `json:"items"`
+}
+
+// +k8s:openapi-gen=true
+// TidbAutoScalerSpec describes the
+type TidbClusterAutoScalerSpec struct {
+	// TidbClusterRef
+	Cluster TidbClusterRef `json:"cluster"`
+
+	// We used prometheus to fetch the metrics resources until the pd could provide it.
+	// MetricsUrl represents the url to fetch the metrics info
+	// +optional
+	MetricsUrl *string `json:"metricsUrl,omitempty"`
```

Comment:
The user may not know the metricsURL

Reply:
Currently, we have to scrape cluster metrics from Prometheus and it is recommended from `TidbMonitor`.  `MetricsUrl` is to provide the service URL of Prometheus. Do you have any suggestion about name?

I will also add explanation and examples for `MetricsUrl` when for `TidbClusterAutoscaler`

Reply:
I mean the operator can get the monitor url automatically from `tidb-monitor` service?

Reply:
In our own case, This could work. But For Operator users, we can't guarantee this. Some users may deploy monitor by helm charts or `TidbMonitor`. Some users may use our scrape config in their own Prometheus. Some users may didn't deploy monitor.

Reply:
I think the issue here is how users know which metric to use and how the MetricUrl to be configured.

Reply:
The users needn't care which metric to use if they use our monitor or copy our job config. And the metricUrl is the Prometheus service url which would be pointed out in documents and examples.